pkg/apis/migration/v1alpha1/verify.go

```diff
@@ -0,0 +1,248 @@
+type Workload struct {
+	Name      string   `json:"name"`
+	Resources []string `json:"resources,omitempty"`
+}
+
+// UnhealthyNamespace is a store for unhealthy resources in a namespace
+type UnhealthyNamespace struct {
+	Name      string     `json:"name"`
+	Workloads []Workload `json:"workloads"`
+}
+
+// IsHealthy will report if there are any unhealthy namespaces were present
+func (m *MigMigration) IsHealthy() bool {
+	return m.Status.UnhealthyResources.Namespaces == nil || len(m.Status.UnhealthyResources.Namespaces) == 0
+}
+
+// VerifyPods would collect, display and notify about the pod health in a specific namespace
+func (u *UnhealthyResources) VerifyPods(client k8sclient.Client, options *k8sclient.ListOptions) (*[]unstructured.Unstructured, error) {
+	podList := corev1.PodList{}
+
+	err := client.List(context.TODO(), options, &podList)
+	if err != nil {
+		return nil, err
+	}
+
+	unhealthy := []unstructured.Unstructured{}
+	for _, pod := range podList.Items {
+		if pod.Status.Phase != corev1.PodFailed && !containerUnhealthy(pod) {
+			continue
+		}
+
+		obj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&pod)
+		if err != nil {
+			return nil, err
+		}
+
+		unstructuredPod := unstructured.Unstructured{Object: obj}
+		unstructuredPod.SetKind("Pod")
+		unhealthy = append(unhealthy, unstructuredPod)
+	}
+
+	return &unhealthy, nil
+}
+
+func containerUnhealthy(pod corev1.Pod) bool {
+	// Consider suspicious containers as an error state
+	podContainersStatuses := append(pod.Status.InitContainerStatuses, pod.Status.ContainerStatuses...)
+	for _, containerStatus := range podContainersStatuses {
+		if containerStatus.Ready {
+			continue
+		}
+		terminationDetails := containerStatus.State.Terminated
+		if terminationDetails != nil && terminationDetails.ExitCode != 0 {
+			return true
+		}
+		waitingDetails := containerStatus.State.Waiting
+		if waitingDetails == nil {
+			continue
+		}
+		switch waitingDetails.Reason {
+		case ImagePullBackOff, CrashLoopBackOff:
+			break
+		default:
+			continue
+		}
+
+		return true
+	}
+
+	return false
+}
+
+func (u *UnhealthyResources) findResources(namespace string, unhealthyResources ...*[]UnhealthyNamespace) *UnhealthyNamespace {
+	for _, namespaces := range unhealthyResources {
+		for _, unhealthyNamespace := range *namespaces {
+			if unhealthyNamespace.Name == namespace {
+				return &unhealthyNamespace
+			}
+		}
+	}
+	return nil
+}
+
+func (u *UnhealthyResources) findWorkload(resources *UnhealthyNamespace, name string) *Workload {
+	for _, workload := range resources.Workloads {
+		if workload.Name == name {
+			return &workload
+		}
+		for _, subWorkload := range workload.Resources {
+			if subWorkload == name {
+				return &workload
+			}
+		}
+	}
+	return nil
+}
+
+// AddUnhealthyNamespace is adding a workload to selected list of unhealthy namespaces
+func (u *UnhealthyResources) AddUnhealthyNamespace(cluster *[]UnhealthyNamespace, namespace, name string, resourceNames ...string) {
```

Comment:
This method is scoped to the `UnhealthyResources` object.  Anything more than `AddNamespace()` is redundant.